--- Kconfig	2017-11-15 12:09:19.000000000 +0300
+++ a/drivers/video/fbdev/Kconfig	2017-11-11 17:54:12.158415692 +0300
@@ -236,6 +236,16 @@
 comment "Frame buffer hardware drivers"
 	depends on FB
 
+config FB_ALTERA_VIP
+	tristate "Altera VIP Frame Reader framebuffer support"
+	depends on FB
+	select FB_CFB_FILLRECT
+	select FB_CFB_COPYAREA
+	select FB_CFB_IMAGEBLIT
+	help
+	    This driver supports the Altera Video and Image Processing(VIP)
+	    Frame Reader
+
 config FB_GRVGA
 	tristate "Aeroflex Gaisler framebuffer support"
 	depends on FB && SPARC
--- Makefile	2017-11-15 12:09:19.000000000 +0300
+++ a/drivers/video/fbdev/Makefile	2017-11-12 14:00:19.696327978 +0300
@@ -19,6 +19,7 @@
 obj-$(CONFIG_FB_PM2)              += pm2fb.o
 obj-$(CONFIG_FB_PM3)		  += pm3fb.o
 
+obj-$(CONFIG_FB_ALTERA_VIP)	  += altvipfb.o
 obj-$(CONFIG_FB_I740)		  += i740fb.o
 obj-$(CONFIG_FB_MATROX)		  += matrox/
 obj-$(CONFIG_FB_RIVA)		  += riva/
